editor/src/clj/editor/code_completion.clj

```diff
@@ -78,16 +97,21 @@
                              :value \"literal doc\"}
                           - {:type :markdown
                              :value \"markdown doc\"
-                             :base-url URI} (optional URI for resolving links)"
-  [name & {:keys [insert-snippet insert-string display-string type commit-characters detail doc]}]
-  {:pre [(not (and insert-string insert-snippet))]
-   :post [(s/assert ::completion %)]}
+                             :base-url URI} (optional URI for resolving links)
+    :tags                 a set of tag keywords contextualizing the completion,
+                          currently the only supported tag is :deprecated"
+  [name & {:keys [insert additional-edits display-string type commit-characters detail doc tags]}]
+  {:post [(s/assert ::completion %)]}
   (cond-> {:name name
            :display-string (or display-string name)
            :insert (cond
-                     insert-snippet {:type :snippet :value insert-snippet}
-                     insert-string {:type :plaintext :value insert-string}
+                     (string? insert) {:type :snippet :value insert}
+                     insert insert
                      :else {:type :plaintext :value name})}
+          tags
+          (assoc :tags tags)
```

Comment:
To be honest, I don't see an issue here. We only use tags set to check if it `contains?` the `:deprecated` tag. We could, for example, make it so that `:tags` is a required key, so we could rely on it and do `(tags :deprecated)` instead of `(contains? tags :deprecated)`. I originally did it that way, but then I thought that `contains?` conveys the intent better. But I don't relly see the need for knowing if the tag set is empty or not.